tests/Unit/Impl/RestStreamxClientTest.php

```diff
@@ -450,6 +452,40 @@ public function shouldThrowServiceFailureExceptionWhenErrorResponseWithSuccessHt
         $this->createPublisherWithIrrelevantSchema("errors")->publish("key", $data);
     }
 
+    /** @test */
+    public function shouldHandleMultiMessageRequestsAndResponses()
+    {
+        // Given
+        $inputMessages = [
+            Message::newPublishMessage('key-1', new Data('Data 1'))->build(),
+            Message::newUnpublishMessage('key-2')->build(),
+            Message::newPublishMessage('key-3', new Data('Data 3'))->build(),
+            Message::newUnpublishMessage('key-4')->build(),
+            Message::newPublishMessage('key-5', new Data('Data 5'))->build()
+        ];
+
+        $messageResponses = [
+            new FailureResponse('SOME_ERROR_CODE', 'Something happened'),
+            new SuccessResult(123, 'key-2'),
+            new SuccessResult(345, 'key-3'),
+            new FailureResponse('SOME_OTHER_ERROR_CODE', 'Something bad happened'),
+            new SuccessResult(567, 'key-5'),
+        ];
```

Comment:
A _random_ mix of publish/unpublish messages and successes/failures while processing them - on purpose to test all cases